ros/src/computing/planning/decision/packages/decision_maker/nodes/decision_maker/decision_maker_node_state_drive.cpp

```diff
@@ -0,0 +1,404 @@
+#include <decision_maker_node.hpp>
+
+namespace decision_maker
+{
+void DecisionMakerNode::updateWaitEngageState(cstring_t& state_name, int status)
+{
+}
+
+void DecisionMakerNode::entryDriveState(cstring_t& state_name, int status)
+{
+}
+
+void DecisionMakerNode::updateDriveState(cstring_t& state_name, int status)
+{
+  if (isArrivedGoal())
+  {
+    tryNextState("arrived_goal");
+    return;
+  }
+
+  if (current_status_.closest_waypoint == -1)
+  {
+    publishOperatorHelpMessage("The vehicle passed last waypoint or waypoint does not exist near the vehicle.");
+    tryNextState("mission_aborted");
+    return;
+  }
+
+  if (isVehicleOnLaneArea())
+  {
+    tryNextState("on_lane_area");
+  }
+  else
+  {
+    tryNextState("on_free_area");
+  }
+}
+
+uint8_t DecisionMakerNode::getSteeringStateFromWaypoint(void)
+{
+  static const double distance_to_target = param_num_of_steer_behind_;
+  static const size_t ignore_idx = 0;
+
+  double distance = 0.0;
+  geometry_msgs::Pose prev_pose = current_status_.pose;
+  uint8_t state = 0;
+
+  if (ignore_idx > current_status_.finalwaypoints.waypoints.size())
+  {
+    return 0;
+  }
+
+  for (unsigned int idx = 0; idx < current_status_.finalwaypoints.waypoints.size() - 1; idx++)
+  {
+    distance += amathutils::find_distance(prev_pose, current_status_.finalwaypoints.waypoints.at(idx).pose.pose);
+
+    state = current_status_.finalwaypoints.waypoints.at(idx).wpstate.steering_state;
+
+    if (state && (state != autoware_msgs::WaypointState::STR_STRAIGHT || distance >= distance_to_target))
+    {
+      break;
+    }
+    prev_pose = current_status_.finalwaypoints.waypoints.at(idx).pose.pose;
+  }
+  return state;
+}
+uint8_t DecisionMakerNode::getEventStateFromWaypoint(void)
+{
+  static const double distance_to_target = param_num_of_steer_behind_;
+  static const size_t ignore_idx = 0;
+
+  double distance = 0.0;
+  geometry_msgs::Pose prev_pose = current_status_.pose;
+  uint8_t state = 0;
+
+  if (ignore_idx > current_status_.finalwaypoints.waypoints.size())
+  {
+    return 0;
+  }
+
+  for (unsigned int idx = 0; idx < current_status_.finalwaypoints.waypoints.size() - 1; idx++)
+  {
+    distance += amathutils::find_distance(prev_pose, current_status_.finalwaypoints.waypoints.at(idx).pose.pose);
+
+    state = current_status_.finalwaypoints.waypoints.at(idx).wpstate.event_state;
+
+    if (state && distance >= distance_to_target)
+    {
+      break;
+    }
+    prev_pose = current_status_.finalwaypoints.waypoints.at(idx).pose.pose;
+  }
+  return state;
+}
+std::pair<uint8_t, int> DecisionMakerNode::getStopSignStateFromWaypoint(void)
+{
+  static const size_t ignore_idx = 0;
+  static const double mu = 0.7;  // dry ground/ asphalt/ normal tire
```

Comment:
I think that the distance determined by this parameter is not the distance to stop the vehicle, but the distance to start recognizing the existence of stop signs.
The distance at which the vehicle actually starts to stop is determined by the parameter of the velocity_set node.

Therefore, I think that there is no problem if the distance calculated by using this parameter is sufficiently large.

Reply:
Yes, I also think that there is no problem if that assumption is correct. If "mu" and "reaction_time" has a large value, it may not be necessary to parameterize.
However, as there is no constraint on the waypoint interval, I think that it can not assume a sufficiently large value for "reset_count".

Reply:
That parameter is used in special situations that the vehicle stops at the stop line and moves backward without crossing the stop line.
In this case, when the vehicle moves backward by the number of waypoints of "reset_count", the stop line is recognized again.

Therefore, it may be a problem if the waypoint interval is extremely large, but is there a need to consider it?

Reply:
I think whether there is a problem if it is realistic with the waypoint interval. (0.1 to 1.0 m)
However, super parameters should be shared not only to the creator but also to other developers.

Reply:
OK.
So then, should I only parameterize reset_cout?

Reply:
Yes, I would like to make it a ROS parameter.
Meanwhile, since I think that management of this parameter is almost unnecessary, it is not necessary to include it in the launch file.

Reply:
I parameterized it with the name `stopline_reset_count`.